### ISSUE TITLE
pull method to get total progress info ?

### ISSUE THREAD

**Issue by iorilu** [ollama/ollama-python#81]
I checked the pull progress example , it seems to return few progresss info , is it possible to get total progress info?

for example i need to make a ui progress bar to reflect the total download progress status

**Comment by mxyng** [comment]
I'm not sure what you mean. The pull example uses streaming to receive progress updates. Each response returns the progress (current and total) in bytes for the current layer being downloaded or a status update.

**Comment by iorilu** [comment]
> I'm not sure what you mean. The pull example uses streaming to receive progress updates. Each response returns the progress (current and total) in bytes for the current layer being downloaded or a status update.

> i mean to get total model download progress info, not the progress of one file, i know it will download a few files for a model

**Comment by BrainSlugs83** [comment]
Yeah, for example, one model might contain 4 files, and the pull status tells you bytes x/y -- but not which file you're on.  It just goes from 0 to 100%, over and over until it's done, and you don't know how many times it will plan to do that.

Maybe there's a way to pull the file count & sizes from the ollama website, but I don't know what that method is...
